### PR TITLE
fix: multichain blocks in serverless hub

### PR DESCRIPTION
**Pull request by Reinis-FRP**
**Motivation**

The logic for multichain bot block number fetching in serverless hub still deviates from single chain bots that result in not caching of fetched block numbers for multichain bots and starting from genesis block first time when multichain bot is added for new chain.

**Summary**

Fixes the issue by using the same logic in multichain block number fetching as is used for single chain bots.

**Details**

Batches single chain and multichain block number fetching in a single promises array that is then processed using the same logic.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [x]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

**Issue(s)**

Fixes: https://linear.app/uma/issue/UMA-2413/improve-block-number-fetching-for-multichain-blocks-in-hub

### CODE REVIEW

File: packages/serverless-orchestration/src/ServerlessHub.js [hunk 1]
```diff
@@ -145,60 +145,67 @@ hub.post("/", async (req, res) => {
...
-      // for the chosen node.
-      if (!lastQueriedBlockNumber && latestBlockNumber) {
-        lastQueriedBlockNumber = latestBlockNumber;
-      }
-      // If the last queried number is still undefined at this point, then exit with an error.
-      else if (!lastQueriedBlockNumber)
-        throw new Error(
-          `No block number for chain ID stored on GCP and cannot read head block from node! chainID:${chainId} spokeCustomNodeUrl:${spokeCustomNodeUrl}`
-        );
+      const blockNumberResults = await Promise.all(blockNumberPromises);
```
Comment (daywiss) [inline packages/serverless-orchestration/src/ServerlessHub.js:185]: is it possible to continue if some of these calls fail, or should teh whole thing fail if any fail?  also does this call strain providers with many chains or should we be OK?

File: packages/serverless-orchestration/test/ServerlessHub.js [hunk 2]
```diff
@@ -771,9 +769,7 @@ describe("ServerlessHub.js", function () {
     const hubConfig = {
       testServerlessBot: {
         serverlessCommand: "echo single network bot started",
-        environmentVariables: {
-          CUSTOM_NODE_URL: network.config.url,
-        },
+        environmentVariables: { CUSTOM_NODE_URL: network.config.url },
```
Comment (daywiss) [inline packages/serverless-orchestration/test/ServerlessHub.js:772]: was formatting run here?

**Review by daywiss — APPROVED** [review]
looks good, minor comments

File: packages/serverless-orchestration/test/ServerlessHub.js (hunk 2 above)
Comment (Reinis-FRP) [reply packages/serverless-orchestration/test/ServerlessHub.js:772]: husky pre-commit auto formatted this

File: packages/serverless-orchestration/test/ServerlessHub.js [hunk 3]
```diff
@@ -798,7 +794,7 @@ describe("ServerlessHub.js", function () {
     // Logs should include correct starting and latest block numbers for the alternate network.
     const alternateBlockNumbers = {
       [alternateChainId]: {
-        lastQueriedBlockNumber: 0,
+        lastQueriedBlockNumber: latestAlternateBlockNumber, // Same as latest as this is first time bot is run.
```
Comment (md0x) [inline packages/serverless-orchestration/test/ServerlessHub.js:797]: Why are they the same the first time the bot is run?

Comment (Reinis-FRP) [reply packages/serverless-orchestration/test/ServerlessHub.js:797]: otherwise the first run would cover all events since genesis.

**Review by md0x — APPROVED** [review]

**Review by mrice32 — APPROVED** [review]
Nice simplification -- agreed that all block number querying/coaching should work the same way.

**Pull request merged**